src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets

```diff
@@ -272,50 +243,30 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!--
     ============================================================
-                                        _FilterOutFXFromResolvedCandidates
-    ============================================================
-    -->
-  
-  <Target Name="_FilterOutFXFromResolvedCandidates"
-          Condition="$(SkipRemovingSystemFiles) != 'true' ">
-
-    <ItemGroup>
-      <_ManagedResolvedFilesToOptimize Include="@(_ManagedResolvedFileToPublishCandidates)" Condition="!Exists('$(_NetCoreRefDir)$(DirectorySeparatorChar)%(FileName)%(Extension)')"/>
-    </ItemGroup>
-  </Target>
-
-  <!--
-    ============================================================
-                                        _PopulateUnFilteredResolvedCandidates
-    ============================================================
-    -->
-  <Target Name="_PopulateUnFilteredResolvedCandidates"
-          Condition="$(SkipRemovingSystemFiles) == 'true' ">
-
-   <ItemGroup>
-      <_ManagedResolvedFilesToOptimize Include="@(_ManagedResolvedFileToPublishCandidates)" />
-    </ItemGroup>
-  </Target>
-
-  <!--
-    ============================================================
                                         _RestoreCrossgen
 Restores netcoreapp and publishes it to a temp directory
     ============================================================
     -->
   
   <Target Name="_RestoreCrossgen"
-           Condition="$(SkipOptimization) != 'true' ">
-    <MSBuild Projects="$(_CrossProjFile)"
+          DependsOnTargets="PrepforRestoreForComposeStore;
```

Comment:
Restore crossgen is only done once to get the crossgen infra setup. The other crossGen is to actually run the crossgen